### PR TITLE
Add a check for the surface if it is valid

### PR DESCRIPTION
**Pull request by mahmuttaskiran**
Fixes an issue where the Surface is not valid and the `draw` method is crashing.

https://github.com/flutter/flutter/issues/155018

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by matanlurey — APPROVED** [review]

File: shell/platform/android/test/io/flutter/plugin/platform/PlatformViewWrapperTest.java [hunk 1]
```diff
@@ -63,6 +65,41 @@ public void onDraw(Canvas canvas) {
...
+    when(surface.isValid()).thenReturn(true);
+    final PlatformViewRenderTarget renderTarget = mock(PlatformViewRenderTarget.class);
+    when(renderTarget.getSurface()).thenReturn(surface);
+    when(surface.lockHardwareCanvas()).thenReturn(canvas);
+    final PlatformViewWrapper wrapper = new PlatformViewWrapper(ctx, renderTarget);
+
+    // Test.
+    wrapper.draw(canvas);
+
+    // Verify.
```
Comment (jonahwilliams) [inline shell/platform/android/test/io/flutter/plugin/platform/PlatformViewWrapperTest.java:99]:
```suggestion
```

**Comment by pushpendraKh** [comment]
In which flutter version we can expect this fix?

**Comment by HongZhiQing** [comment]
> In which flutter version we can expect this fix?

Same question.

**Comment by mahmuttaskiran** [comment]
It was released in 3.24.5 
https://github.com/flutter/flutter/blob/stable/CHANGELOG.md#3245

**Pull request merged**